AdminClient/src/app/orders/order-details-dialog/order-details-dialog.component.html

```diff
@@ -0,0 +1,106 @@
+<h1 mat-dialog-title>{{data.title}}</h1>
+<mat-dialog-content>
+  <form
+    [formGroup]="orderForm"
+    class="max-height"
+  >
+    <div fxLayout="row">
+      <mat-form-field>
+        <mat-label>Order Date</mat-label>
+        <input matInput [matDatepicker]="orderDatePicker" formControlName="orderDate">
+        <mat-datepicker-toggle matSuffix [for]="orderDatePicker"></mat-datepicker-toggle>
+        <mat-datepicker #orderDatePicker></mat-datepicker>
+        <mat-error *ngIf="dateHasError" id="dateError">{{dateErrorMessage}}</mat-error>
+      </mat-form-field>
+    </div>
+    <div fxLayout="row">
+      <mat-form-field>
+        <mat-label>Order Status</mat-label>
+        <input matInput formControlName="status">
+        <mat-error *ngIf="statusHasError" id="statusError">{{statusErrorMessage}}</mat-error>
+      </mat-form-field>
+    </div>
+    <div fxLayout="row">
+      <mat-form-field>
+        <mat-label>Representative Email</mat-label>
+        <mat-select formControlName="user" id="user">
+          <mat-option *ngFor="let user of repUsers" [value]="user">{{ user.email }}</mat-option>
+        </mat-select>
+        <mat-error *ngIf="userHasError" id="userError">{{userErrorMessage}}</mat-error>
+      </mat-form-field>
+    </div>
+    <div fxLayout="row" *ngIf="editable">
+      <button (click)="addNewEntry()" color="primary" mat-flat-button>Add another product to this order</button>
+    </div>
+    <div formArrayName="orderProducts">
+      <div fxLayout="row"
+           *ngFor="let orderProduct of orderProducts.controls; let index = index"
+           [formGroupName]="index">
+        <div fxLayout="column">
+          <mat-form-field>
+            <mat-label>Quantity</mat-label>
+            <input matInput type="number" formControlName="quantity">
+            <mat-error *ngIf="quantityHasError(index)" id="quantityError">{{quantityErrorMessage(index)}}</mat-error>
+          </mat-form-field>
+        </div>
+        <div fxLayout="column">
+          <mat-form-field>
+            <mat-label>Product</mat-label>
+            <mat-select formControlName="product" id="product">
+              <mat-option
+                *ngFor="let product of products" [value]="product"
+              >
+                {{ product.name }}
+              </mat-option>
+            </mat-select>
```

Comment:
Have you tested this in case of order edit?

Reply:
Yes I have